ProcessLib/HydroMechanics/HydroMechanicsFEM.h

```diff
@@ -0,0 +1,425 @@
+        storage_p.setZero(pressure_size, pressure_size);
+
+        typename ShapeMatricesType::template MatrixType<displacement_size,
+                                                        pressure_size>
+            Kup;
+        Kup.setZero(displacement_size, pressure_size);
+
+        double const& dt = _process_data.dt;
+
+        SpatialPosition x_position;
+        x_position.setElementID(_element.getID());
+
+        unsigned const n_integration_points =
+            _integration_method.getNumberOfPoints();
+        for (unsigned ip = 0; ip < n_integration_points; ip++)
+        {
+            x_position.setIntegrationPoint(ip);
+            auto const& w = _ip_data[ip].integration_weight;
+
+            auto const& N_p = _ip_data[ip]._N_p;
+            auto const& dNdx_p = _ip_data[ip]._dNdx_p;
+
+            auto const& B = _ip_data[ip]._b_matrices;
+            auto const& sigma = _ip_data[ip]._sigma;
+
+            auto const& C = _ip_data[ip]._C;
+
+            double const S =
+                _process_data.storage_coefficient(t, x_position)[0];
+            double const K_over_mu =
+                _process_data.intrinsic_permeability(t, x_position)[0] /
+                _process_data.fluid_viscosity(t, x_position)[0];
+            auto const alpha = _process_data.biot_coefficient(t, x_position)[0];
+            auto const rho_sr = _process_data.solid_density(t, x_position)[0];
+            auto const rho_fr = _process_data.fluid_density(t, x_position)[0];
+            auto const porosity = _process_data.porosity(t, x_position)[0];
+            auto const body_force =
+                _process_data.specific_body_force(t, x_position);
+            assert(body_force.size() == DisplacementDim);
+            auto const b =
+                Eigen::Map<typename ShapeMatricesType::template VectorType<
+                    DisplacementDim> const>(body_force.data(), DisplacementDim);
+
+            auto const& identity2 = MaterialLib::SolidModels::Invariants<
+                kelvin_vector_size>::identity2;
+
+            //
+            // displacement equation, displacement part
+            //
+            _ip_data[ip].updateConstitutiveRelation(t, x_position, dt, u);
+
+            local_Jac
+                .template block<displacement_size, displacement_size>(
+                    displacement_index, displacement_index)
+                .noalias() += B.transpose() * C * B * w;
+
+            typename ShapeMatricesType::template MatrixType<DisplacementDim,
+                                                            displacement_size>
+                N_u = ShapeMatricesType::template MatrixType<
+                    DisplacementDim,
+                    displacement_size>::Zero(DisplacementDim,
+                                             displacement_size);
+            for (int i = 0; i < DisplacementDim; ++i)
+                N_u.template block<1, displacement_size / DisplacementDim>(
```

Comment:
Storing laplace_p and storage_p in the class is more expensive (in time) than creation in the function.
Storing the large N_u in the integration points gains some speedup.
(Tested on square 1e2 example running approximately 6.7s)